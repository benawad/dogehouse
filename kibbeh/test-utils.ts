import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
