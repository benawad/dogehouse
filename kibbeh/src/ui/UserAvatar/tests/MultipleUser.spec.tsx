import React from "react";

import { render } from "../../../../test-utils";
import { MultipleUsers } from "../";

describe("UserAvatar", () => {
  describe("MultipleUsers", () => {
    it("should render 3 single users max.", () => {
      const srcArray = ["", "", "", ""];
      const { getAllByTestId } = render(<MultipleUsers srcArray={srcArray} />);
      const avatar = getAllByTestId("single-user-avatar");

      expect(avatar).toHaveLength(3);
    });

    it("should render 2 single users", () => {
      const srcArray = ["", ""];
      const { getAllByTestId } = render(<MultipleUsers srcArray={srcArray} />);
      const avatar = getAllByTestId("single-user-avatar");

      expect(avatar).toHaveLength(2);
    });

    it("should match snapshot", () => {
      const srcArray = ["", "", "", ""];

      const { getAllByTestId } = render(<MultipleUsers srcArray={srcArray} />);
      const avatar = getAllByTestId("single-user-avatar");

      expect(avatar).toMatchSnapshot();
    });
  });
});
