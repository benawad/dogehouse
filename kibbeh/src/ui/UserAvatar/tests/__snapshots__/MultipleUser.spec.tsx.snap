// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar MultipleUsers should match snapshot 1`] = `
Array [
  <div
    class="relative inline-block "
    data-testid="single-user-avatar"
    style="width: 20px; height: 20px;"
  >
    <img
      alt="your-avatar"
      class="rounded-full w-full h-full object-cover "
      src=""
    />
  </div>,
  <div
    class="relative inline-block "
    data-testid="single-user-avatar"
    style="width: 20px; height: 20px;"
  >
    <img
      alt="your-avatar"
      class="rounded-full w-full h-full object-cover "
      src=""
    />
  </div>,
  <div
    class="relative inline-block "
    data-testid="single-user-avatar"
    style="width: 20px; height: 20px;"
  >
    <img
      alt="your-avatar"
      class="rounded-full w-full h-full object-cover "
      src=""
    />
  </div>,
]
`;
